crates/bcr-ebill-transport/src/nostr.rs

```diff
@@ -400,46 +415,125 @@ impl TransportClientApi for NostrClient {
+    async fn publish_metadata(&self, node_id: &NodeId, data: &Metadata) -> Result<()> {
+        // Get the signer for this identity
+        let signer = self.get_signer(node_id)?;
+
+        // Build and sign the metadata event with the specific identity
+        let event = EventBuilder::metadata(data)
+            .build(signer.public_key())
+            .sign(&*signer)
+            .await
+            .map_err(|e| {
+                error!("Failed to sign metadata event: {e}");
+                Error::Crypto("Failed to sign metadata event".to_string())
+            })?;
+
+        self.client().await?.send_event(&event).await.map_err(|e| {
             error!("Failed to send user metadata with Nostr client: {e}");
             Error::Network("Failed to send user metadata with Nostr client".to_string())
         })?;
         Ok(())
     }
 
-    async fn publish_relay_list(&self, relays: Vec<RelayUrl>) -> Result<()> {
-        self.update_relay_list(relays.into_iter().map(|r| (r, None)).collect())
+    async fn publish_relay_list(&self, node_id: &NodeId, relays: Vec<RelayUrl>) -> Result<()> {
+        // Get the signer for this identity
+        let signer = self.get_signer(node_id)?;
+
+        // Build and sign the relay list event with the specific identity
+        let relay_list: Vec<(RelayUrl, Option<RelayMetadata>)> =
+            relays.into_iter().map(|r| (r, None)).collect();
+        let event = EventBuilder::relay_list(relay_list)
+            .build(signer.public_key())
+            .sign(&*signer)
             .await
             .map_err(|e| {
-                error!("Failed to send relay list with Nostr client: {e}");
-                Error::Network("Failed to send relay list with Nostr client".to_string())
+                error!("Failed to sign relay list event: {e}");
+                Error::Crypto("Failed to sign relay list event".to_string())
             })?;
+
+        self.client().await?.send_event(&event).await.map_err(|e| {
+            error!("Failed to send relay list with Nostr client: {e}");
+            Error::Network("Failed to send relay list with Nostr client".to_string())
+        })?;
         Ok(())
     }
 
     async fn connect(&self) -> Result<()> {
-        if !self.connected.load(Ordering::Relaxed) {
-            self.connected.store(true, Ordering::Relaxed);
+        if self
+            .connected
+            .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+            .is_ok()
+        {
             self.client.connect().await;
-            self.publish_relay_list(self.config.relays.clone()).await?;
+
+            // Publish relay list for ALL identities
+            let node_ids = self.get_all_node_ids();
+            let relay_urls: Vec<RelayUrl> = self
+                .relays
+                .iter()
+                .filter_map(|r| RelayUrl::parse(r.as_str()).ok())
+                .collect();
+
+            for node_id in node_ids {
```

Comment:
if this is slow/blocking, it might be something to do in a spawned task, or concurrently in the future

Reply:
good idea